Add documentation for `Rule.use` as a function (#2694)
webpack/webpack.js.org: src/content/configuration/module.md

## `Rule.use`

`[UseEntry] | function`

`Rule.use` can be an array of [UseEntry](#useentry) which are applied to modules. Each entry specifies a loader to be used.

Passing a string (i.e. `use: [ 'style-loader' ]`) is a shortcut to the loader property (i.e. `use: [ { loader: 'style-loader '} ]`).

Loaders can be chained by passing multiple loaders, which will be applied from right to left (last to first configured).

`Rule.use` can be a function which receives the object argument describing the module being loaded, and must return an array of `UseEntry`.

The same shortcut as an array can be used for the return value (i.e. `use: [ 'style-loader' ]`).

The object parameter has the following fields:
- `resource`: The path to the module being loaded
- `realResource`: The path to the module being loaded

> - `resourceQuery`: ??

Reviewer: ```suggestion
 - `resourceQuery`
```

Reviewer: Or investigate what it is

## `Rule.use`

`[UseEntry] | function`

`Rule.use` can be an array of [UseEntry](#useentry) which are applied to modules. Each entry specifies a loader to be used.

Passing a string (i.e. `use: [ 'style-loader' ]`) is a shortcut to the loader property (i.e. `use: [ { loader: 'style-loader '} ]`).

Loaders can be chained by passing multiple loaders, which will be applied from right to left (last to first configured).

`Rule.use` can be a function which receives the object argument describing the module being loaded, and must return an array of `UseEntry`.

The object parameter has the following fields:

- `compiler`: The current webpack compiler (can be undefined)
- `issuer`: The path to the module that is importing the module being loaded
- `resource`: The path to the module being loaded

> - `realResource`: The path to the module being loaded

Reviewer: It is worth to note what differs `resource` and `realResource` here as it may confuse the reader @bew

Author: Indeed that would be better, but as stated in the comments above, I don't actually know what is the difference ^^
If you can help about this I'd be very grateful :)

Reviewer 2: `resource` is usually equal `realResource` expect when the `resource` name is overwritten via `!=!` in request string. `realResource` is always the resource file which is requested.

`require("LLL!RRR")` -> `resource: RRR, realResource: RRR`
`require("MMM!=!LLL!RRR")` -> `resource: MMM, realResource: RRR`

Author: Thanks, I'll add a note when I'm at work

## `Rule.use`

> `array`

Reviewer: ```suggestion
`[UseEntry]` `function`
```

Reviewer: to be consistent with the rest of the docs

> An array of [UseEntry](#useentry) which are applied to modules. Each entry specifies a loader to be used.

Reviewer: ```suggestion
`Rule.use` can be an array of [UseEntry](#useentry) which are applied to modules. Each entry specifies a loader to be used.
```

Passing a string (i.e. `use: [ 'style-loader' ]`) is a shortcut to the loader property (i.e. `use: [ { loader: 'style-loader '} ]`).

Loaders can be chained by passing multiple loaders, which will be applied from right to left (last to first configured).

> `Function`

Reviewer: ```suggestion
```

> A function which takes an object describing the module being loaded, and must return an array of `UseEntry`.

Reviewer: ```suggestion
`Rule.use` can be a function which receives the object argument describing the module being loaded, and must return an array of `UseEntry`.
```

Although i am not sure it must return an array or single UseEntry

> The same shortcut as as an array can be used for the return value (i.e. `use: [ 'style-loader' ]`).

Reviewer: ```suggestion
The same shortcut as an array can be used for the return value (i.e. `use: [ 'style-loader' ]`).
```

> The object parameter has the following fields:

Reviewer: Move this info above the previous statement about shortcut. Please investigate question marks, wouldn't want to merge todos
